crates/loader/src/local/mod.rs

```diff
@@ -71,6 +71,17 @@ async fn prepare(
 ) -> Result<Application<CoreComponent>> {
     let info = info(raw.info, &src);
 
+    let id_counts: HashMap<String, u8> =
+        raw.components.iter().fold(HashMap::new(), |mut counts, c| {
+            *counts.entry(c.id.clone()).or_insert(0) += 1;
+            counts
+        });
+    for (id, count) in id_counts {
```

Comment:
This version works now, I'm going through some Rust docs to figure out a way to make this cleaner. Ideally, I'd avoid the extra `for` loop, and maybe avoid creating a new `HashMap` here.